src/structures/GuildEvent.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const Base = require('./Base');
+const { PrivacyLevels, GuildEventEntityTypes, GuildEventStatuses } = require('../util/Constants');
+
+/**
+ * Represents an event in a {@link Guild}.
+ * @extends {Base}
+ */
+class GuildEvent extends Base {
+  /**
+   *
+   * @param {Client} client The instantiating client
+   * @param {*} data The data for the guild event
```

Comment:
Once the upstream PR lands, please add the `@external` API docs link (maybe add a TODO comment so you don't forget)